readthedocs/api/v2/views/model_views.py

```diff
@@ -66,6 +66,49 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         return data.encode(self.charset)
 
 
+class DisableListEndpoint:
+
+    """
+    Helper to disable APIv2 listing endpoint.
+
+    We are disablng the listing endpoint because it could cause DOS without
+    using any type of filtering.
+
+    This class disables these endpoints except:
+
+     - version resource when passing ``?project__slug=``
+     - build resource when using ``?commit=``
+
+    All the other type of listings are disabled and return 409 CONFLICT with an
+    error message pointing the user to APIv3.
+    """
+
+    def list(self, *args, **kwargs):
+        disabled = True
+
+        # NOTE: keep list endpoint that specifies a resource
+        if any([
+                self.basename == 'version' and 'project__slug' in self.request.GET,
+                self.basename == 'build' and 'commit' in self.request.GET,
+        ]):
+            disabled = False
+
+        if not disabled:
+            return super().list(*args, **kwargs)
+
+        return Response(
+            {
+                'error': 'disabled',
+                'msg': (
+                    'List endpoint have been disabled: please provide a resource identifier. '
```

Comment:
```suggestion
                    'List endpoint have been disabled due to heavy resource usage: please provide a resource identifier. '
```

I have no idea what a resource identifier is. We should provide examples or something.

Reply:
A resource identifier is the object pk, or commit, or the unique id we use for that particular object: `/api/v2/projects/` -> `/api/v2/projects/1/`

We may want to remove that part of the message tho and just point to APIv2.

Reply:
I changed the message and remove the part of the resource id.